src/core/exec-invoke.c

```diff
@@ -4890,6 +4931,27 @@ int exec_invoke(
                 }
         }
 
+        if (context->private_pids) {
+                if (ns_type_supported(NAMESPACE_PID)) {
+                        r = setup_private_pids(context, params);
+                        if (ERRNO_IS_NEG_PRIVILEGE(r))
+                                log_exec_warning_errno(context, params, r,
+                                                       "PrivatePIDs=yes is configured, but pid namespace setup failed, ignoring: %m");
+                        else if (r < 0) {
+                                *exit_status = EXIT_NAMESPACE;
+                                return log_exec_error_errno(context, params, r, "Failed to set up pid namespace: %m");
+                        }
+
+                        if (r > 0)
+                                /* If we're in the parent process, return immediately in order to exit ASAP.
+                                 * The rest of the setup will continue in the child process. */
```

Comment:
maybe make the comment clearer: start with saying "Return value > 0 means we are in the parent process"